internal/services/sentinel/azuresdkhacks/dataconnectors.go

```diff
@@ -0,0 +1,162 @@
+package azuresdkhacks
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/validation"
+	securityinsight "github.com/tombuildsstuff/kermit/sdk/securityinsights/2022-10-01-preview/securityinsights"
+)
+
+type DataConnectorsClient struct {
```

Comment:
could you describe this hack and why it is required?

Reply:
That is described at: https://github.com/hashicorp/terraform-provider-azurerm/pull/19209/files#diff-678a173979b1ea700c88ac1db19d4545fe5088817771414efcaaacafbad2dd5aR13

Reply:
@magodo mind sending a Swagger PR to fix that too, in addition to the issue?

Reply:
@tombuildsstuff That is a service behavior issue rather than a Swagger issue, which the service team has acknowledged and will fix it at backend.

Reply:
any updates on if the service team has fixed this?

Reply:
@katbyte Checking with the service team...

Reply:
@katbyte The service team told that they'll revisit it at Apirl next year.

Reply:
@magodo - could you update the linked rest api spec issue with that? will this hack here work when that backend change goes into play?

also could we put a `//todo 4.0 check if this can be removed" comment next to the link to the rest issue?

thanks

Reply:
@katbyte The issue link is added here: https://github.com/hashicorp/terraform-provider-azurerm/blob/e39e85256f9156e2e8b8d88efa3f911f3ad2681d/internal/services/sentinel/azuresdkhacks/models.go#L13. I'll add the todo above it then.

Reply:
Is there any way that you could write the hack so it can handle both? otherwise when the fix goes out the resource will break users requiring them to update again

Reply:
@katbyte I've made the change and now it should go on working even if the service team fixed it:

```shell
terraform-provider-azurerm on  sentinel_dc_titaxii via 🐹 v1.19 took 6s
💢  TF_ACC=1 go test -v -timeout=20h ./internal/services/sentinel -run=TestAccDataConnectorThreatIntelligenceTAXII_basic
=== RUN   TestAccDataConnectorThreatIntelligenceTAXII_basic
=== PAUSE TestAccDataConnectorThreatIntelligenceTAXII_basic
=== CONT  TestAccDataConnectorThreatIntelligenceTAXII_basic
--- PASS: TestAccDataConnectorThreatIntelligenceTAXII_basic (262.73s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel      262.743s
```

Reply:
perfect! thanks 😄